application/tests/extraobjects_test.yaml

```diff
@@ -0,0 +1,190 @@
+suite: ExtraObjects
+
+templates:
+  - extraobjects.yaml
+
+tests:
+  - it: does not yield objects if extraObjects is []
+    set:
+      extraObjects: []
+    asserts:
+      - hasDocuments:
+          count: 0
+
+  - it: yields one ConfigMap (list, object)
+    set:
+      extraObjects:
+        - apiVersion: v1
+          kind: ConfigMap
+          metadata:
+            name: test-config
+            namespace: test-namespace
+          data:
+            key1: value1
+            key2: '{{ print "value2" }}'
+    asserts:
+      - hasDocuments:
+          count: 1
+      - isKind:
+          of: ConfigMap
+      - equal:
+          path: metadata.name
+          value: test-config
+      - equal:
+          path: metadata.namespace
+          value: test-namespace
+      - equal:
+          path: data.key1
+          value: value1
+      - equal:
+          path: data.key2
+          value: value2
+
+  - it: yields one ConfigMap (list, string)
+    set:
+      extraObjects:
+        - |
+          apiVersion: v1
+          kind: ConfigMap
+          metadata:
+            name: test-config
+            namespace: test-namespace
+          data:
+            key1: value1
+            key2: '{{ print "value2" }}'
+    asserts:
+      - hasDocuments:
+          count: 1
+      - isKind:
+          of: ConfigMap
+      - equal:
+          path: metadata.name
+          value: test-config
+      - equal:
+          path: metadata.namespace
+          value: test-namespace
+      - equal:
+          path: data.key1
+          value: value1
+      - equal:
+          path: data.key2
+          value: value2
+
+  - it: yields one ConfigMap (object, string)
```

Comment:
I believe this test is wrong, it should use a map as extraObjects (from the name test name)

Reply:
https://github.com/stakater/application/pull/382/commits/41ddefd39181fb1fd53b27b172bb15cc2666bd50